hearth/lib/hearth/auth_schemes.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Hearth
+  module AuthSchemes
+    # Base class for all AuthScheme classes.
+    class Base
+      def initialize(scheme_id:)
+        @scheme_id = scheme_id
+      end
+
+      # @return [String]
+      attr_reader :scheme_id
+
+      # @return [IdentityResolver, nil]
+      def identity_resolver(identity_resolvers = {})
+        identity_resolvers[@identity_type]
+      end
+
+      # @return [Signers::Base]
+      attr_reader :signer
```

Comment:
Is there a reason we don't have an attr_reader for identity_type?

Reply:
It's never accessed publicly. It's only referenced internally when calling identity_resolver with a hash of possible identity resolvers.